Feature/250 api v4

alexiglesias31 opened the pull request:
### Changes proposed in this Pull Request:

Closes #250

#

- Upgrade ads/v3 to ads/v4 endpoints.
- Use new catalog APIs for merchant registration, domain verification, and catalog ingestion.


### Screenshots:


### Detailed test instructions:
1. Install the plugin.
2. Make the connection.
3. Everything should work as usual.

### Changelog entry

>

tomalec committed:
Fire `get_started_faq` tracking event,

When clicking on getting started page faq item to collapse or expand it.

Install `@woocommerce/tracks` locally, for linting and testing. It's externalized with DEWP.

Addresses a part of https://github.com/woocommerce/pinterest-for-woocommerce/issues/232

tomalec committed:
Provide `question_id` to `wcadmin_pfw_get_started_faq` event,

Adhere to Tracks Event naming convention.

tomalec committed:
Document the `wcadmin_pfw_get_started_faq` event.

tomalec committed:
Add JSDoc template for trackign events .md,

Add a plugin to resolve .~ alias.

Related to
- https://github.com/woocommerce/pinterest-for-woocommerce/issues/232
- https://github.com/woocommerce/google-listings-and-ads/issues/117

tomalec committed:
Install JSDoc, with `jsdoc-plugin-typescript`,

to run on our imported types.

Related to
- https://github.com/woocommerce/pinterest-for-woocommerce/issues/232
- https://github.com/woocommerce/google-listings-and-ads/issues/117

tomalec committed:
Fire Track Events on Account connection step.

- `wcadmin_pfw_account_connect_button_click`
- `wcadmin_pfw_account_create_button_click`
From https://github.com/woocommerce/pinterest-for-woocommerce/issues/232
Plus additional:
- `wcadmin_pfw_account_convert_button_click`
- `wcadmin_pfw_account_disconnect_button_click`
- `wcadmin_pfw_business_account_create_button_click`
- `wcadmin_pfw_business_account_connect_button_click`

tomalec committed:
Remove duplicated `wcadmin_` from event name,

addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/237#discussion_r742623832.

tomalec committed:
Merge pull request #237 from woocommerce/add/232-tracks

tomalec committed:
Remove duplicated `wcadmin_` from event name,

addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/237#discussion_r742623832.

tomalec committed:
Fix indentation,

remove not used plugin from example config,
use JS syntax for highlighting JSON with comments in `woo-tracking-jsdoc` files.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/238#pullrequestreview-797483220

tomalec committed:
Escape `\` in JSON example in `woo-tracking-jsdoc/README.md`

Co-authored-by: Eason Su <eason.su.tw@gmail.com>

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/238#discussion_r742655507

tomalec committed:
Add the `woo-tracking-jsdoc` layer to the `templates` config in `woo-tracking-jsdoc/README.md`

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/238#discussion_r742670060

tomalec committed:
Make the `woo-tracking-jsdoc` disclaimer agnostic to its `replacement`.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/238#discussion_r742688638

tomalec committed:
Make `woo-tracking-jsdoc`'s replacement match attributes.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/238#discussion_r742694967

Co-authored-by: Eason Su <eason.su.tw@gmail.com>

tomalec committed:
Add a new line between every Trackign Event docs.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/238#discussion_r742733381.

tomalec committed:
Merge pull request #239 from woocommerce/add/232-account-track

tomalec committed:
Escape `\` in `woo-tracking-jsdoc/README.md`

Co-authored-by: Eason <eason.su.tw@gmail.com>

tomalec committed:
Merge remote-tracking branch 'origin/feature/232-tracking' into add/232-jsdoc-tracks

tomalec committed:
Generate tracking docs for the new events,

added in the merged `feature/232-tracking` branch.

tomalec committed:
Merge pull request #238 from woocommerce/add/232-jsdoc-tracks

tomalec committed:
Tweak Tracks docs generation,

- sort events by name
- add sourcefile lineno to the link
- add line break after the end disclaimer
- support empty descriptions

tomalec committed:
Document lins of Track event emitters.

tomalec committed:
Fix a typo in code comment woo-tracking-jsdoc/publish.js

Co-authored-by: Eason <eason.su.tw@gmail.com>

tomalec committed:
Fix whitespaces in `woo-tracking-jsdoc/publish.js`.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/254#pullrequestreview-805853439

tomalec committed:
Match multiline `@fires` descriptions.

Addresses a bit of https://github.com/woocommerce/pinterest-for-woocommerce/pull/254#discussion_r749191354

tomalec committed:
Create JSDoc plugin to support `fires`' description

Address https://github.com/woocommerce/pinterest-for-woocommerce/pull/254#pullrequestreview-805853439

tomalec committed:
Merge branch 'develop' into feature/232-tracking

tomalec committed:
Add Track Events to documentation links.

Change `StepOverview` to accept more props for read more `Button` link.

tomalec committed:
Merge pull request #254 from woocommerce/feature/232-jsdoc-tweaks

tomalec committed:
Merge branch 'develop' into feature/232-tracking

tomalec committed:
Merge branch 'feature/232-tracking' into feature/232-docs

tomalec committed:
Add Track Event for merchant guidelines link.

After https://github.com/woocommerce/pinterest-for-woocommerce/pull/255 was merged to `feature/232-tracking`

tomalec committed:
Separate Documentation Link props creation to a helper.

Make the code more DRY, and assert consistency for all links.

tomalec committed:
Set `rel:'noopener'` for external link Buttons.

To secure them in IE.

tomalec committed:
Make `onClick` optional for `documentation-link-props`

Co-authored-by: Eason <eason.su.tw@gmail.com>

tomalec committed:
Add the missing prefix to `pfw_documentation_link_click`.

In `documentation-link-props` JSDoc

Co-authored-by: Eason <eason.su.tw@gmail.com>

tomalec committed:
Merge pull request #261 from woocommerce/feature/232-docs

tomalec committed:
Make `documentationLinkProps` use `rel: noopener` by default.

Alternative take for https://github.com/woocommerce/pinterest-for-woocommerce/pull/266#discussion_r751907452

tomalec committed:
Add a note in JSDoc about overwriting `onClick`

to somewhat mitigate https://github.com/woocommerce/pinterest-for-woocommerce/pull/266#discussion_r751915176

tomalec committed:
Update TRACKING.md docs.

alexiglesias31 committed:
Remove unnecesary fields on create merchant method

tomalec committed:
Merge pull request #266 from woocommerce/feature/232-docs-props

puntope committed:
Add missing link in Setup Tracking section

alexiglesias31 committed:
Completely remove unnecessary args from Merchants::get_merchant method

ksere committed:
remove unneeded variable

puntope committed:
Merge pull request #279 from woocommerce/fix/262-fix-setup-tracking-link

Add missing link in Setup Tracking section

tomalec committed:
Merge branch 'develop' into feature/232-tracking

eason9487 committed:
Add the shared constants for general process/label status

eason9487 committed:
Implement a StatusLabel component to render a text with a prefix graphic marking

eason9487 committed:
Implement a UrlInputControl component to Renders an InputControl with a link icon prefix

eason9487 committed:
Pass the store URL containing the protocol to frontend.

eason9487 committed:
Use <Notice> to show the message of site validation failure.

eason9487 committed:
Implement the UI of formatted site URL and adjust verification button

eason9487 committed:
Remove the unused component StepStatus

eason9487 committed:
Add JSDoc, simplify variable and replace hardcode strings with constants for ClaimWebsite

eason9487 committed:
Fix CSS lint errors

tomalec committed:
Merge branch 'develop' into feature/232-tracking

tomalec committed:
Support multiline Track Event properties' description.

tomalec committed:
Fire track event for modal,

implements part of https://github.com/woocommerce/pinterest-for-woocommerce/issues/232

alexiglesias31 committed:
Api version 4

alexiglesias31 committed:
[Updated] Create tag method

alexiglesias31 committed:
[Updated] Get advertiser tags method

eason9487 committed:
Remove unnecessary wildcard selectors and fix an incorrect selector

Address
- https://github.com/woocommerce/pinterest-for-woocommerce/pull/286#discussion_r763022659
- https://github.com/woocommerce/pinterest-for-woocommerce/pull/286#discussion_r763023548

Co-authored-by: Gan Eng Chin <ecgan@users.noreply.github.com>

eason9487 committed:
Revert an irrelevant replacement of hardcoded string.

Address https://github.com/woocommerce/pinterest-for-woocommerce/pull/286#discussion_r763008750

Co-authored-by: Gan Eng Chin <ecgan@users.noreply.github.com>

alexiglesias31 committed:
[Updated] Get merchant feed method

alexiglesias31 committed:
Get feed report method

eason9487 committed:
Use the default background color of `<Notice>` component.

Address https://github.com/woocommerce/pinterest-for-woocommerce/pull/286#discussion_r763027686

alexiglesias31 committed:
[Added] Disconnect merchant method

alexiglesias31 committed:
[Fixed] disconnect merchant endpoint

eason9487 committed:
Merge pull request #286 from woocommerce/tweak/110-site-domain-format

Tweak the UI of claim website

puntope committed:
Track pfw_save_changes_button_click

alexiglesias31 committed:
Get tag id from response frontend

alexiglesias31 committed:
Single method to create/update merchant and feed

alexiglesias31 committed:
v4 endpoints used only for ads API

puntope committed:
Refactoring and tests

puntope committed:
Added docs for tracking docs generation

puntope committed:
Change function description

alexiglesias31 committed:
[Fixed] Delete data oonly if merchant is disconnected sucessfully

alexiglesias31 committed:
Check merchant status on feed_state check

alexiglesias31 committed:
Namespace for ProductFeedStatus on API\Base

alexiglesias31 committed:
[Added] optional add json content-type on request headers

alexiglesias31 committed:
[Added] Advertisers connect and disconnect methods

alexiglesias31 committed:
Added throw description Pinterest_For_Woocommerce::disconnect()

alexiglesias31 committed:
Spell error on Api\Base::disconnect_merchant method name

alexiglesias31 committed:
Force json encode request body ads API

alexiglesias31 committed:
[Fixed] Tags got empty on connect advertiser method

tomalec committed:
Fix the copy in `wcadmin_pfw_modal_closed` event docs,

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/290#discussion_r763756953

tomalec committed:
Allow multipel new-lines in event property description,

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/290#discussion_r763778061

Co-authored-by: Eason Su <eason.su.tw@gmail.com>

puntope committed:
Added eventName support in documentation-link-props.js

puntope committed:
Added links in class-pinterest-for-woocommerce-admin.php

puntope committed:
Adding test for PreLaunch Notice record Event

puntope committed:
Adding some tests for PreLaunchNotice component

puntope committed:
Added required link in jest config

puntope committed:
Track get_started_notice_link_click on PrelaunchNotice click

puntope committed:
Adding @woocommerce/settings dependency (DEWP) in Jest Config.

puntope committed:
Adding tracking in UnsupportedCountryNotice component

puntope committed:
Improve tracking docs

puntope committed:
Fix linter errors, use spaces instead of tabs

alexiglesias31 committed:
Merge branch 'develop'

Conflicts:
	src/API/Base.php
	src/API/FeedState.php
	src/ProductSync.php

alexiglesias31 committed:
Fix add_feed_registration_state method after merge develop

alexiglesias31 committed:
Fix API\Base after merge develop

alexiglesias31 committed:
[Fixed] PHP coding standards

alexiglesias31 committed:
Removed unneeded comment on Merchants::get_merchant()

alexiglesias31 committed:
Typing error on Base::disconnect_advertiser

alexiglesias31 committed:
Refactor Merchants::get_merchant_object

alexiglesias31 committed:
Added wrapper method Merchants::update_or_create_merchant

alexiglesias31 committed:
Added Feeds class to get feed by id and location

alexiglesias31 committed:
Add some extra comments to code

alexiglesias31 committed:
[Added] New status Queued for processing

puntope committed:
Adding better exceptions in DomainVerification.php

puntope committed:
Adding exception status mapping

puntope committed:
Add tracking when domain verification fails

puntope committed:
FIX PHP CS warnings

puntope committed:
Add tracking docs

ksere commented on class-pinterest-for-woocommerce.php:
```diff
@@ -566,22 +567,37 @@ public static function save_token( $token ) {
-			delete_option( PINTEREST_FOR_WOOCOMMERCE_DATA_NAME );
+			// Disconnect merchant from Pinterest.
+			try {
+				$result = Pinterest\API\Base::disconnect_merchant();
```
Not sure but maybe we also need to disconnect the advertiser. 
The doc says: 

> A required step when changing advertisers **or removing an account** to disconnect advertiser and tag from your platform on Pinterest.

ksere commented on src/API/AdvertiserConnect.php:
```diff
@@ -0,0 +1,120 @@
+				throw new \Exception( esc_html__( 'Missing advertiser or tag parameters.', 'pinterest-for-woocommerce' ), 400 );
+			}
+
+			$connected_advertiser = Pinterest_For_Woocommerce()::get_data( 'connected_advertiser' );
```
I think this can be simplified and not save duplicate stuff into the db. We already save `tracking_advertiser` & `tracking_tag` in options.

ksere commented on src/API/AdvertiserConnect.php:
```diff
@@ -0,0 +1,120 @@
+	 */
+	private static function disconnect_advertiser( $connected_advertiser, $connected_tag ) {
+
+		if ( $connected_advertiser && $connected_tag ) {
```
This conditional would be better in `connect_advertiser()`

ksere commented on src/API/Base.php:
```diff
@@ -66,17 +68,18 @@ public static function instance() {
 	 * @throws \Exception PHP exception.
 	 */
-	public static function make_request( $endpoint, $method = 'POST', $payload = array(), $api = '', $cache_expiry = false ) {
+	public static function make_request( $endpoint, $method = 'POST', $payload = array(), $api = '', $cache_expiry = false, $is_json_payload = false ) {
```
I would add a condition as we do for the API version, which would add the header to the request directly if possible. 
Eg. something like:
```
if ( 'ads/' === $api && 'POST' === $method ) {
$request['headers']['Content-Type'] = 'application/json';
}

ksere commented on src/API/FeedState.php:
```diff
@@ -146,7 +146,7 @@ public function get_feed_state() {
 
 			/* Translators: The error description as returned from the API */
-			$error_message = sprintf( esc_html__( 'Error getting feed\'s state. [%s]', 'pinterest-for-woocommerce' ), $th->getMessage() );
+			$error_message = sprintf( esc_html__( 'Error getting feed state. [%s]', 'pinterest-for-woocommerce' ), $th->getMessage() );
```
was that wrong?

ksere commented on src/Feeds.php:
```diff
@@ -0,0 +1,107 @@
+	 * @throws \Exception PHP Exception.
+	 */
+	public static function get_merchant_feed( $merchant_id, $feed_id ) {
+		try {
```
```suggestion

		try {
```

ksere commented on src/Feeds.php:
```diff
@@ -0,0 +1,107 @@
+}
+
+/**
+ * Class Handling registration & generation of the XML product feed.
```
Stale comment from other class

ksere commented on src/Feeds.php:
```diff
@@ -0,0 +1,107 @@
+	 * @throws \Exception PHP Exception.
+	 */
+	public static function get_merchant_feed_by_location( $merchant_id, $feed_location ) {
+		try {
```
```suggestion

		try {
```

ksere commented on src/API/Base.php:
```diff
@@ -399,108 +465,21 @@ public static function get_merchant( $merchant_id ) {
 
 	/**
-	 * Creates a merchant for the authenticated user or returns the existing one.
+	 * Creates a merchant for the authenticated user or updates the existing one.
```
```suggestion
	 * Creates a merchant for the authenticated user or updates the existing one. On success the Merchant ID is returned instead of the full merchant object. 
```

ksere commented on src/ProductSync.php:
```diff
@@ -507,97 +495,61 @@ public static function trigger_async_feed_registration_asap() {
-				if ( $configured_path === $local_path && $feed_args['country'] === $product_pin_feed_profile->country && $feed_args['locale'] === $product_pin_feed_profile->locale ) {
-					// We can assume we're on the same site.
+				// The response only contains the merchant id.
+				$response = Merchants::update_or_create_merchant();
```
Do we need to run this again here since its already run in the `get_merchant()` in line [511](https://github.com/saucal/pinterest-for-woocommerce/pull/7/files#diff-9eb8ac4928cebc9cc9e52da272589aafa0747ad026d7f101c0f9d6b9951af91fR511) ?

As I understand, especially on a new account `::update_or_create_merchant();` will be called in `Merchants::get_merchant();` and also again here. What's the case we are trying to catch ?

alexiglesias31 committed:
Use already registered settings tracking_advertiser and tracking_tag

alexiglesias31 committed:
Refactor conditional on AdvertiserConnect::disconnect_advertiser

alexiglesias31 committed:
Disconnect advertiser too on Pinterest_For_Woocommerce::disconnect

alexiglesias31 committed:
Conditional to force json content-type header on ads API

alexiglesias31 committed:
Fixed class comment on Feeds

alexiglesias31 committed:
Improved comment for method Base::update_or_create_merchant

alexiglesias31 committed:
Fixed phpcs on API::Base

alexiglesias31 committed:
Removed unneeded extra parameter on Base::make_request calls

alexiglesias31 committed:
Simplify logic to force json content-type header

alexiglesias31 committed:
Replace "" surrounded endpoints by ''

ksere commented on src/ProductSync.php:
```diff
@@ -507,97 +495,61 @@ public static function trigger_async_feed_registration_asap() {
-			if ( false === $product_pin_feed_profile_id ) {
-				$configured_path = dirname( $product_pin_feed_profile->location_config->full_feed_fetch_location );
-				$local_path      = dirname( $feed_args['feed_location'] );
+			// Check if there is a registered feed id.
+			$prev_registered = self::get_registered_feed_id();
+
+			// Update feed if there was no one registered previously.
+			if ( ! $prev_registered ) {
```
```suggestion
			// Update feed if we don't have a feed_id saved.
			// for cases where the already existed in the API.
			if ( ! self::get_registered_feed_id() ) {
```

alexiglesias31 committed:
Removed unnecessary false parameter

alexiglesias31 committed:
Merge branch 'feature/250-api-v4' of https://github.com/saucal/pinterest-for-woocommerce into feature/250-api-v4

alexiglesias31 committed:
Simplify get registered feed on ProductSync::register_feed

Co-authored-by: Kostas Seresiotis <k.seresiotis@gmail.com>

alexiglesias31 committed:
Spell error on error message

ksere committed:
Merge pull request #294 from woocommerce/fix/196-unused-parameters-sent-request-create-merchant

Fix/196 unused parameters sent request create merchant

puntope committed:
Adding tests for domain verification failure

(cherry picked and modified from commit c095c414374ea76e6c1210ba7b6987958eafc573
- `tests/tracking.test.js` was renamed to `setup-guide/app/steps/ClaimWebsite.test.js` )
Co-authored-by: Miguel Perez Pellicer <5908855+puntope@users.noreply.github.com>

puntope committed:
Adding default state value for updatedData

puntope committed:
Adding unit tests for tracking functionality

puntope committed:
Adding Usage Tracker functionality in the plugin

puntope committed:
Initialize the Tracker on plugin init.

alexiglesias31 committed:
Connected advertiser message only triggered if advertiser or tag change

alexiglesias31 committed:
Force wp_json_encode method in all POST request to ads API

puntope committed:
Update context logic in Save Settings Button recordEvent

puntope committed:
Remove updatedData state property for tracking

puntope committed:
Remove view in ConnectionApp.js and SettingsApp.js and use constant for the name

puntope committed:
Update Tracking docs

alexiglesias31 committed:
[Fixed] Feed was being registered twice on merchant

alexiglesias31 committed:
Verify that local feed is properly registered in the API

alexiglesias31 committed:
Fixed undefined error on tag creation response

puntope committed:
Update documentation regarding saveToDb optional parameter.

Co-authored-by: Eason <eason.su.tw@gmail.com>

tomalec committed:
Set different context for `pfw_modal_*` and `pfw_documentation_link_click`

Track Events, in settings and wizard respectively.
Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/290#pullrequestreview-824948568.

puntope committed:
Update assets/source/setup-guide/app/components/SaveSettingsButton.js

Co-authored-by: Tomek Wytrębowicz <tomalecpub@gmail.com>

alexiglesias31 committed:
Refactor connect advertiser method

alexiglesias31 committed:
Connect advertiser if needed after a plugin upgrade

alexiglesias31 committed:
Added translators error description to logged message

tomalec committed:
Add a basic test for `wcadmin_pfw_modal_open`.

alexiglesias31 committed:
Fixed tracking_advertiser and tracking_tag are settings not data

tomalec committed:
Add tests for `pfw_modal_closed`,

Add `@testing-library/user-event` devDependency.

alexiglesias31 committed:
Disconnect advertiser on hook update_option

tomalec committed:
Merge pull request #290 from woocommerce/add/232-modal

alexiglesias31 committed:
Merge branch 'develop' into feature/250-api-v4

puntope committed:
Add View param on SaveSettingsButton

puntope committed:
Rolled back unnecessary return

puntope committed:
Update tests

puntope committed:
Remove prefix & fix typo in docs

puntope committed:
Add missing quotes

puntope committed:
Rollback to @link instead of @see

puntope committed:
Fix JSDOC comment

(cherry picked from commit 7ea85eacea5235392c76c3080405733468aea7c5)

puntope committed:
Update documentation for tracking

Co-authored-by: Tomek Wytrębowicz <tomalecpub@gmail.com>

puntope committed:
Update documentation for tracking in UnsupportedCountryNotice

Co-authored-by: Tomek Wytrębowicz <tomalecpub@gmail.com>

puntope committed:
Remove deprecated documentation comment

Co-authored-by: Tomek Wytrębowicz <tomalecpub@gmail.com>

puntope committed:
Remove deprecated todo in prelaunch-notice

puntope committed:
Update tracking docs in SaveSettingsButton

Co-authored-by: Tomek Wytrębowicz <tomalecpub@gmail.com>

puntope committed:
Merge pull request #295 from woocommerce/add/232-save-event

Track Save Button Click

tomalec committed:
Merge branch 'develop' into feature/232-tracking

tomalec committed:
Merge branch 'feature/232-tracking' into add/232-notice-link-click

tomalec committed:
Re-generate tracking docs, after the merge.

`npm run doc:tracking`

tomalec committed:
Merge pull request #299 from woocommerce/add/232-notice-link-click

tomalec committed:
Add `wcadmin_pfw_setup` tracking events.

Implments part of https://github.com/woocommerce/pinterest-for-woocommerce/issues/232,
with slightly changed properties as described in https://github.com/woocommerce/pinterest-for-woocommerce/issues/232#issuecomment-998968676.

tomalec committed:
Make `@woocomerce/settings` mock export named `getSettings`.

To mimic the original module behavior.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/issues/306#issuecomment-999151843.

tomalec committed:
Add `toContainURLSearchParam` - custom jest matcher.

So we could easily assert WP-admin navigation.

tomalec committed:
Add tests for `wcadmin_pfw_setup` fired from the LandingPage.

tomalec committed:
Merge branch 'feature/232-tracking' into add/232-verify-domain-failed-separated-tests

puntope committed:
Fixing test after merge

puntope committed:
Merge pull request #300 from woocommerce/add/232-verify-domain-failed

Add tracking for verify domain failed

puntope committed:
Refactor tests

puntope committed:
Remove parenthesis in export default

tomalec committed:
Merge branch 'feature/232-tracking' into add/232-setup-event

tomalec committed:
Merge pull request #308 from woocommerce/add/232-setup-event

tomalec committed:
Refactor `AccountConnection` component.

- Show the Pinterest logo, regardless of the state, to reduce the number of moving parts.
- Show the "Disconnect" button even if there is no account data, or before it's loaded. To help users disconnect even if some account data gets corrupted, or simply without waiting for everything to load.
- Show placeholder instead of a spinner when we wait only for the account details.
- Set the minimum height of the card's content to `40px` = the height of the spinner, to reduce the difference between states, and reduce the layout shift when the spinner disappears and the content is loaded.
- -Make the code DRY, by moving repeated markup outside of conditions.

puntope committed:
Refactor Jest Config mocking

puntope committed:
Merge remote-tracking branch 'origin/feature/232-tracking' into add/232-refactor-testing-tracking

# Conflicts:
#	assets/source/tests/dependencies/woocommerce/settings.js

tomalec committed:
Add context property to `wcadmin_pfw_account_disconnect_button_click` event,

to avoid the need of a separate `disconnected_account_link_click` one.

Implements part of https://github.com/woocommerce/pinterest-for-woocommerce/issues/232.

puntope committed:
Merge pull request #311 from woocommerce/add/232-refactor-testing-tracking

Refactor testing tracking

tomalec committed:
Fix CSS linting - whitespace issues

tomalec committed:
Merge pull request #313 from woocommerce/add/232-disconnect-track

tomalec committed:
Remove useless `calc` from CSS.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/312#discussion_r776151383

tomalec committed:
Fix `pfw_domain_verify_failure` tests,

Mock ESM default export, make the mock sync.

tomalec committed:
Fire `wcadmin_pfw_domain_verify_success` event,

implements part of https://github.com/woocommerce/pinterest-for-woocommerce/issues/232

tomalec committed:
Re-generate TRACKING.md docs

tomalec committed:
Replace `setTimout` with `waitFor`

in `pfw_domain_verify_success` test.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/314#discussion_r776989581

tomalec committed:
Remove redundant `@woocommerce/tracks` mock.

It's already mocked by `jest.config`'s `moduleNameMapper`.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/314#discussion_r776990038

tomalec committed:
Merge pull request #314 from woocommerce/add/232-verify-success

alexiglesias31 committed:
Call AdvertiserConnect::disconnect_advertiser instead Base

alexiglesias31 committed:
Import WP_Error on AdvertiserConnect

alexiglesias31 committed:
Use double quotes and variable for strings instead . concatenation

alexiglesias31 committed:
Import classes to avoid prefixes in code

alexiglesias31 committed:
Return empty string if local feed is not registered

alexiglesias31 committed:
Fixed return type from mixed to object in methods description

alexiglesias31 committed:
Simplify get merchant object logic

alexiglesias31 committed:
Import AdvertiserConnect class to avoid prefix namespace

alexiglesias31 committed:
Use POST for connect advertiser request

puntope committed:
Merge pull request #301 from woocommerce/feature/291-woo-tracker

Adding Woo Tracker for Usage Tracking

alexiglesias31 committed:
[Fixed] Connect advertiser on complete setup or save settings

alexiglesias31 committed:
Refactor connectAdvertiser method

tomalec committed:
Add `pfw_ads_manager_link_click` track event.

Addresses https://github.com/woocommerce/pinterest-for-woocommerce/pull/296#pullrequestreview-842872047

tomalec committed:
Merge pull request #316 from woocommerce/add/232-ads-manager

tomalec committed:
Merge pull request #296 from woocommerce/feature/232-tracking

tomalec committed:
Merge pull request #312 from woocommerce/update/account-connection

alexiglesias31 committed:
Merge branch 'develop' into feature/250-api-v4

alexiglesias31 committed:
Remove empty lines between imports

Co-authored-by: Gan Eng Chin <ecgan@users.noreply.github.com>

alexiglesias31 committed:
Use try-catch on complete setup action

alexiglesias31 committed:
Evaluate result instead result.reconnected

Co-authored-by: Gan Eng Chin <ecgan@users.noreply.github.com>

alexiglesias31 committed:
Place connect advertiser method in a separate file

alexiglesias31 committed:
Remove trailing slash tagowner route

alexiglesias31 committed:
Return reconnected parameter on connectAdvertiser method
